misc/releaser.py

```diff
@@ -109,36 +130,44 @@ def __str__(self):
         return base
 
     @property
-    def is_preversion(self):
+    def is_preversion(self) -> bool:
         return self.pre_type is not None
 
-    def __eq__(self, other):
-        if type(other) == tuple:
-            other = (
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Version):
+            string_other = str(other)
+        elif isinstance(other, tuple):
+            string_other = (
                 "v"
                 + ".".join([str(i) for i in other[:3]])
                 + (f"-{str().join(other[3:])}" if len(other) > 3 else "")
             )
-        return str(self) == str(other)
+        else:
+            raise TypeError(f"Unsupported type `{type(other).__name__}`")
+
+        return str(self) == string_other
 
-    def __lt__(self, other):
+    def __lt__(self, other: Version) -> bool:
         k = (self.major, self.minor, self.patch)
         other_k = (other.major, other.minor, other.patch)
 
-        def _pre_type_to_val(pre_type):
+        def _pre_type_to_val(pre_type: str) -> int:
             return {"a": 1, "b": 2, "rc": 3}[pre_type]
 
         if k == other_k:
             # Must take into account dev and pre info
             if self.is_preversion:
-                pre_type = _pre_type_to_val(self.pre_type)
-                pre_index = self.pre_index
+                # Mypy doesn't understand that we already checked the `self.pre_type != None` in `is_preversion`
```

Comment:
add a
```python
assert self.pre_type is not None
assert self.pre_index is not None
```

this is cheap and fix the typing issue ^^

Reply:
done